packages/babel-parser/src/plugins/typescript/index.ts

```diff
@@ -4379,6 +4378,70 @@ export default (superClass: ClassWithMixin<typeof Parser, IJSXParserMixin>) =>
         );
       }
     }
+
+    /**
+     * This hook is defined in the ESTree plugin.
+     * The TS-ESLint always define optional AST properties, here we provide the
+     * default value for such properties immediately after `finishNode` was invoked.
+     *
+     * @param node The AST node finished by finishNode
+     * @returns
+     */
+    fillOptionalPropertiesForTSESLint(node: N.Node): void {
+      switch (node.type) {
+        case "SpreadElement":
+        case "RestElement":
+          node.value = undefined;
```

Comment:
The `.value` is never used by our parser, so we simply set it to `undefined`.

Reply:
What does typescript-eslint use it for?

Reply:
https://github.com/typescript-eslint/typescript-eslint/blob/f0072dc5f00fc24815fc25ad0e645412c1b84f38/packages/ast-spec/src/parameter/RestElement/spec.ts#L8-L15

The `value` is an `AssignmentPattern` or `undefined`. But typescript-eslint always put the nested node to `argument` for these examples:

```js
var [...x = 1] = [];
({ ...x = {} } = {});
```

I guess the `.value` here is a placeholder property such that the typing of `object.properties.map(x => x.value)` can work without guarding the last rest element. 

@JoshuaKGoldberg Could you shred some light here?